app/Console/Commands/Takedate.php

```diff
@@ -45,22 +50,36 @@ public function handle(Extractor $metadataExtractor)
 		$i = $from;
 		foreach ($photos as $photo) {
 			$url = Storage::path('big/' . $photo->url);
-			if (file_exists($url)) {
-				$info = $metadataExtractor->extract($url, $photo->type);
-				if ($info['takestamp'] != null) {
-					$photo->takestamp = $info['takestamp'];
-					if ($photo->save()) {
-						$this->line($i . ': Takestamp updated for ' . $photo->title);
-					} else {
-						$this->line($i . ': Failed to update takestamp for ' . $photo->title);
-					}
+			$i++;
+			if (!file_exists($url)) {
+				$this->line($i . ': File does not exist for ' . $photo->title . '.');
+				continue;
+			}
+			$info = $metadataExtractor->extract($url, $photo->type);
+			$stamp = $info['takestamp'];
+			if ($stamp != null) {
+				$photo->takestamp = $stamp;
+				if ($photo->save()) {
+					$this->line($i . ': Takestamp updated for ' . $photo->title);
 				} else {
-					$this->line($i . ': Could not get Takestamp data for ' . $photo->title . '.');
+					$this->line($i . ': Failed to update takestamp for ' . $photo->title);
 				}
+				continue;
+			}
+			if (!$timestamps) {
+				$this->line($i . ': Could not get Takestamp data for ' . $photo->title . '.');
+				continue;
+			}
+			$filename = $url;
```

Comment:
why is this needed? `$url` is not used anymore afterwards.

if you do not like `$url` then just rename all the previous `$url` before by `$filename`.